src/playground-project.ts

```diff
@@ -51,6 +51,20 @@ const generateUniqueSessionId = (): string => {
   return sessionId;
 };
 
+declare global {
+  interface HTMLElementEventMap {
+    filesChanged: FilesChangedEvent;
+  }
+}
+
+export class FilesChangedEvent extends Event {
+  projectLoaded: boolean;
```

Comment:
WDYT about a separate `projectLoaded` event? Seems like that could be a little simpler.

Reply:
The current work of `filesChanged` needs to happen for both "project loaded" and "project files changed" cases, so it'd be a bit more complex to listen both events and dedupe them correctly. Leaving as is per discussion.